Source/OxyPlot/Rendering/RenderContext/RenderingExtensions.cs

```diff
@@ -93,6 +93,25 @@ M3    M3
                 return;
             }
 
+            if (rc.SetClip(clippingRectangle))
+            {
+                rc.DrawLine(points, stroke, strokeThickness, dashArray, lineJoin, aliased);
+                rc.ResetClip();
```

Comment:
I change the SetClip method in the Wpf.CanvasRenderContext to:

```csharp
        public bool SetClip(OxyRect clippingRect)
        {
            if (this.clip != null)
            {
                throw new InvalidOperationException("SetClip cannot be nested.");
            }

            this.clip = this.ToRect(clippingRect);
            return true;
        }
```

this causes a large number of tests to break.